src/main/java/org/apache/commons/math4/util/FastMathCalc.java

```diff
@@ -99,10 +101,10 @@ private FastMathCalc() {
      * @param TANGENT_TABLE_A table of the most significant part of the tangents
      * @param TANGENT_TABLE_B table of the most significant part of the tangents
      */
-    @SuppressWarnings("unused")
-    private static void buildSinCosTables(double[] SINE_TABLE_A, double[] SINE_TABLE_B,
-                                          double[] COSINE_TABLE_A, double[] COSINE_TABLE_B,
-                                          int SINE_TABLE_LEN, double[] TANGENT_TABLE_A, double[] TANGENT_TABLE_B) {
+    @SuppressWarnings({"unused", "index:array.access.unsafe.high"}) // #1: i is odd as checked by the previous if-else, so i/2 + 1 < SINE_TABLE_LEN
```

Comment:
The fact you've given about `i` doesn't seem to follow from `i` being odd. I think the fact that `i < SINE_TABLE_LEN` (the loop condition) is also necessary, so you should mention it in the comment here.